### PR TITLE
feat: Support S3 transfer acceleration

### PR DESCRIPTION
**Pull request by terrytangyuan**
This PRs adds support for [S3 Transfer acceleration](https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration.html).

### CODE REVIEW

**Comment by sivanantha321** [comment]
Can we expose this configuration in [configmap](https://github.com/kserve/kserve/blob/c52753519e1802e73581856b1ae3a0f2f178d457/config/configmap/inferenceservice.yaml#L167) ?

File: pkg/agent/storage/utils.go [hunk 1]
```diff
@@ -127,10 +127,16 @@ func GetProvider(providers map[Protocol]Provider, protocol Protocol) (Provider,
...
+		useAccelerateString, ok := os.LookupEnv(s3credential.S3UseAccelerate)
+		useAccelerate := false
+		if ok && strings.ToLower(useAccelerateString) == "true" {
+			useAccelerate = true
+		}
 
 		awsConfig := aws.Config{
 			Region:           aws.String(region),
 			S3ForcePathStyle: aws.Bool(!useVirtualBucket),
+			S3UseAccelerate:  aws.Bool(useAccelerate),
```
Comment (yuzisun) [inline pkg/agent/storage/utils.go:139]: how about storage initializer? I do not see code handing here
https://github.com/kserve/kserve/blob/master/python/kserve/kserve/storage/storage.py#L143

Comment (terrytangyuan) [reply pkg/agent/storage/utils.go:139]: Added

**Comment by terrytangyuan** [comment]
> Can we expose this configuration in [configmap](https://github.com/kserve/kserve/blob/c52753519e1802e73581856b1ae3a0f2f178d457/config/configmap/inferenceservice.yaml#L167) ?

Yep just added.

**Pull request merged**